### PR TITLE
lib/json: remove Jsonable and replace it by Serializable

### PR DESCRIPTION
**Pull request by Morriar**
Remove the now useless Jsonable class.

Let's see if something is broken.

Signed-off-by: Alexandre Terrasa <alexandre@moz-code.org>

### CODE REVIEW

File: lib/json/static.nit [hunk 1]
```diff
@@ -152,56 +146,36 @@ redef class FlatText
...
 # A map that can be translated into a JSON object.
-interface JsonMapRead[K: String, V: nullable Jsonable]
+interface JsonMapRead[K: String, V: nullable Serializable]
 	super MapRead[K, V]
-	super Jsonable
+	super Serializable
 end
 
 # A JSON Object.
 class JsonObject
```
Comment (xymus) [inline lib/json/static.nit]: Do you have a usage for this class other than as a shorter alternative to `HashMap[String, nullable Serializable]`?

It's a good reason to keep it, but if it's only as a shortcut, @privat could be interested in replacing it by a virtual type (of the same name) on `Object`.

Comment (privat) [inline lib/json/static.nit]: virtual type on Object is an hack and have issues as is. I prefer to wait to solve there issues.

File: lib/json/static.nit [hunk 2]
```diff
@@ -152,56 +146,36 @@ redef class FlatText
...
-	super JsonMapRead[String, nullable Jsonable]
-	super HashMap[String, nullable Jsonable]
+	super JsonMapRead[String, nullable Serializable]
+	super HashMap[String, nullable Serializable]
 end
 
 # A sequence that can be translated into a JSON array.
-class JsonSequenceRead[E: nullable Jsonable]
-	super Jsonable
+class JsonSequenceRead[E: nullable Serializable]
```
Comment (privat) [inline lib/json/static.nit]: `JsonSequenceRead` -> `SerializableSequenceRead`?

File: lib/json/static.nit (hunk 1 above)
Comment (Morriar) [inline lib/json/static.nit]: I'm not fan of the object Class having a VT used only by one subclass.

Also, I'm really not ready to write `HashMap[String, nullable Serializable]` everywhere. In fact, writing it once in this message was enough. Never again.

Let's keep it this way and reassess in 6 months.

File: lib/json/static.nit (hunk 2 above)
Comment (Morriar) [inline lib/json/static.nit]: `SerializableSequenceRead[E: nullable Serializable]`

Hum... you sure?

Comment (jcbrinfo) [inline lib/json/static.nit]: Hum… that’s a great use case for me… 😏

**Pull request merged**